### DISCUSSION TITLE
queue rate limit to Transactions Per Minute (TPM)

### DISCUSSION THREAD

**Discussion by sibelius** [OptimalBits/bull#2684] [Q&A]
This is not a bug, more an issue to understand how the queue rate limit works in practice

I have a queue like this

```
RATE_LIMITED: new Queue('RATE_LIMITED', config.REDIS_HOST, {
    defaultJobOptions,
    settings,
    limiter: {
      max: 50, // 50 transactions at most
      duration: 10000, // every 10 seconds
    },
  }),
  ```
  
  setting this rate limit config is giving me 250/280 TPM in production
  
  how do you convert TPM to rate limiter config?

**Comment by sibelius** [comment]
can you also expand on bounceBack?

if the rate limit is hit when adding a new job, this new job will go directly to delayed instead of waiting?

**Comment by sibelius** [comment]
what happens if we change the queue rate limit and make a new release ?

the new rate limit is applied to current jobs or new jobs? what about delayed jobs?

**Reply by manast** [reply]
You will find answers to most of your questions here: https://blog.taskforce.sh/rate-limit-recipes-in-nodejs-using-bullmq/
Furthermore, I would recommend you to use BullMQ instead as the newer rate limiter implementation is much more robust.

**Reply by sibelius** [reply]
is there a guide or codemod to migrate from bull to bullmq?

**Reply by manast** [reply]
The recommendation is to start with new queues, other than that the API is quite similar and since you have TS typings fairly straightforward.
